lib/src/Window.cpp

```diff
@@ -43,6 +46,12 @@ vulk::Window::Window(unsigned int width, unsigned int height, const char* title)
     if (!m_windowHandle)
         throw std::runtime_error(utils::getGLFWError());
 
+    m_keyboard = std::make_unique<Keyboard>(Keyboard(m_windowHandle));
+    m_mouse = std::make_unique<Mouse>(Mouse(m_windowHandle));
```

Comment:
Here you call the constructor twice. `make_*` calls the constructor in place in memory.

So in this case, you construct a Keyboard/Mouse, then copy it to a new malloc.  
The fix is just this:
```suggestion
    m_keyboard = std::make_unique<Keyboard>(m_windowHandle);
    m_mouse = std::make_unique<Mouse>(m_windowHandle);
```